### PR TITLE
Automatic Example Collator

### PR DESCRIPTION
**Pull request by coriolinus**
This PR automates the checklist in the README, adding docker files and scripting sufficient to automatically run two relay chain nodes, a collator chain node, extract the genesis state and wasm file, use those files to register the parachain, and verify the collator is continuing to produce new blocks over time.

### CODE REVIEW

File: docker/test-parachain-collator.dockerfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# it busts the cache every time non-source files like inject_bootnodes.sh change,
+# as well as when non-`.dockerignore`'d transient files (*.log and friends)
+# show up. There is no way to exclude particular files, or write a negative
+# rule, using Docker's COPY syntax, which derives from go's filepath.Match rules.
+#
+# We can't combine these into a single big COPY operation like
+# `COPY collator consensus network runtime test Cargo.* .`, because in that case
+# docker will copy the _contents_ of each directory into the image workdir,
+# not the actual directory. We're stuck just enumerating them.
+COPY Cargo.* ./
```
Comment (bkchr) [inline docker/test-parachain-collator.dockerfile]: Just copy all. The moment we add or remove a folder this stuff is broken.

File: docker/test-parachain-collator.dockerfile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+COPY collator collator/
+COPY consensus consensus/
+COPY network network/
+COPY runtime runtime/
+COPY test test/
+
+RUN time cargo build --release -p cumulus-test-parachain-collator
```
Comment (bkchr) [inline docker/test-parachain-collator.dockerfile]:
```suggestion
RUN cargo build --release -p cumulus-test-parachain-collator
```

Cargo already prints the build time.

File: docker/test-parachain-collator.dockerfile (hunk 1 above)
Comment (coriolinus) [reply docker/test-parachain-collator.dockerfile]: Right, will do. During development of the scripts, it really helped that I didn't need to rebuild the rust binary each time (which we do when copying all), but hopefully the scripts are more stable post-merge.

**Comment by zhangtianai** [comment]
> This PR automates the checklist in the README, adding docker files and scripting sufficient to automatically run two relay chain nodes, a collator chain node, extract the genesis state and wasm file, use those files to register the parachain, and verify the collator is continuing to produce new blocks over time.

Is there a detailed description of the steps？

**Comment by JoshOrndorff** [comment]
@zhangtianai There is a tutorial that takes you through spinning up the relay chain and parachains locally. substrate.dev/cumulus-workshop

**Pull request merged**